src/Bass/Shared/Bass/PInvoke/Plugin.cs

```diff
@@ -68,11 +68,25 @@ public static int PluginLoad(string FilePath)
 
             var dir = Path.GetDirectoryName(FilePath);
             var fileName = Path.GetFileName(FilePath);
-            
+
             // Try for Windows, Linux/Android and OSX Libraries respectively.
-            return new[] { $"{dir}{fileName}.dll", $"{dir}lib{fileName}.so", $"{dir}lib{fileName}.dylib" }
-                           .Select(PluginLoad)
-                           .FirstOrDefault(HLib => HLib != 0);
+            string[] paths = new[] {
+                Path.Combine(dir, $"{fileName}.dll"),
+                Path.Combine(dir, $"lib{fileName}.so"),
+                Path.Combine(dir, $"lib{fileName}.dylib")
+            };
+
+            foreach (string path in paths)
+            {
+                // Check if the file exists before trying to load plugin otherwise Bass.LastError can be overwritten by multiple calls.
+                if (File.Exists(path))
+                {
+                    return BASS_PluginLoad(path);
```

Comment:
I believe this would cause problems if all the different platform libraries are put in the same folder.
Like for example on a Mac, this would find the DLL first, try to load the DLL and fail without trying to load the `.dylib`.

Reply:
Yeah that's a good point my use case is in unity and there only the specific platform shared libraries get copied in the build process so I didn't test for this case.

I'll try coming up with a solution for this later today after I get off work.